### ISSUE TITLE
[BUG] final_preds_and_avg_probs.csv file is not generated after inference

### ISSUE THREAD

**Issue by ujjwalbaid0408** [mlcommons/GaNDLF#862]
### To Reproduce
1. Use the data `BreastMNIST from https://medmnist.com/'`
2. Run command `'python /content/GaNDLF/gandlf_run -c /content/config.yaml -i /content/medmnist/dataset/test_path.csv -m /content/model/ -t False -d cuda'`

### Expected behavior
After running inference for the classification task following .csv files must be generated:
- output_predictions.csv
- final_preds_and_avg_probs.csv
- logits.csv

But final_preds_and_avg_probs.csv and logits.csv are not generating.

### Environment information
<!-- Put the output of the following command:
python ./gandlf_debugInfo

      GANDLF version: 0.0.20-dev
      Git hash: 50f369f9671f5b618e9e50568b349a419eca6a65
      Platform: Linux-6.1.58+-x86_64-with-glibc2.35
      Machine: x86_64
      Processor: x86_64
      Architecture: 64bit ELF
      Python environment:
        Version: 3.10.12
        Implementation: CPython
        Compiler: GCC 11.4.0
        Build: main Nov 20 2023 15:14:05

**Comment by sarthakpati** [comment]
Thanks for the report, @ujjwalbaid0408. What are the details about the environment, i.e., the result of `python ./gandlf_debugInfo`?

**Comment by sarthakpati** [comment]
Thanks!

**Comment by sarthakpati** [comment]
@ujjwalbaid0408 can you check if the fix works for you?
